go/vt/vttablet/tabletserver/txthrottler/tx_throttler.go

```diff
@@ -55,8 +56,12 @@ func resetTxThrottlerFactories() {
 	healthCheckFactory = func(topoServer *topo.Server, cell string, cellsToWatch []string) discovery.HealthCheck {
 		return discovery.NewHealthCheck(context.Background(), discovery.DefaultHealthCheckRetryDelay, discovery.DefaultHealthCheckTimeout, topoServer, cell, strings.Join(cellsToWatch, ","))
 	}
-	topologyWatcherFactory = func(topoServer *topo.Server, hc discovery.HealthCheck, cell, keyspace, shard string, refreshInterval time.Duration, topoReadConcurrency int) TopologyWatcherInterface {
-		return discovery.NewCellTabletsWatcher(context.Background(), topoServer, hc, discovery.NewFilterByKeyspace([]string{keyspace}), cell, refreshInterval, true, topoReadConcurrency)
+	topologyWatcherFactory = func(topoServer *topo.Server, hc discovery.HealthCheck, cell, keyspace, shard string, refreshInterval time.Duration, topoReadConcurrency int) (TopologyWatcherInterface, error) {
+		cellTabletsFilter, err := discovery.NewFilterByShard([]string{fmt.Sprintf("%s|%s", keyspace, shard)})
```

Comment:
The conventional way to combine keyspace&shard is with `/`. Is there a reason the above uses `|` to separate them? If there's no particular reason, please accept the below suggestion:

```suggestion
		cellTabletsFilter, err := discovery.NewFilterByShard([]string{fmt.Sprintf("%s/%s", keyspace, shard)})
```

Reply:
@shlomi-noach for some reason [`discovery.NewFilterByShard` uses `|` as a delimiter](https://github.com/vitessio/vitess/blob/main/go/vt/discovery/topology_watcher.go#L319) for filters. It's strange because later in that func [the format you suggest is used in an error](https://github.com/vitessio/vitess/blob/main/go/vt/discovery/topology_watcher.go#L336) 😄